src/Data/HashMap/Linear.hs

```diff
@@ -205,29 +213,36 @@ lookup hmap k = (Unsafe.toLinear lookupFromIx) $ queryIndex hmap k
     lookupFromIx :: (HashMap k v, RobinQuery k) -> (HashMap k v, Maybe v)
     lookupFromIx (h, IndexToInsert _ _) = (h, Nothing)
     lookupFromIx (h@(HashMap _ arr), IndexToUpdate _ ix) =
-      case read arr ix of (_,(_,v,_)) -> (h, Just v)
+      case read arr ix of (_, (_, v, _)) -> (h, Just v)
     lookupFromIx (h, IndexToSwap _ _) = (h, Nothing)
 
-
 -- | Internal function:
 -- Find the index a key ought to hash into, and the PSL it should have.
 -- NOTE: if the underlying array is too small, this never terminates.
 queryIndex :: Keyed k => HashMap k v #-> k -> (HashMap k v, RobinQuery k)
 queryIndex = Unsafe.toLinear unsafeQueryIx
   where
-    unsafeQueryIx :: Keyed k =>
-      HashMap k v -> k -> (HashMap k v, RobinQuery k)
+    unsafeQueryIx :: Keyed k => HashMap k v -> k -> (HashMap k v, RobinQuery k)
     unsafeQueryIx h@(HashMap (size, _) arr) key =
-      (h, walkDownArr (arr, hash key `mod` size) (key, 0) size )
-
+      (h, walkDownArr (arr, (hash key) `mod` size) (key, 0) size)
     walkDownArr :: Keyed k =>
       (RobinArr k v, Int) -> (k, PSL) -> Int -> RobinQuery k
     walkDownArr (arr, ix) (key, psl) size = case read arr ix of
       (_, (_, _, psl_ix)) | psl_ix == (-1) -> IndexToInsert (key, psl) ix
       (_, (k_ix, _, _)) | key == k_ix -> IndexToUpdate (key, psl) ix
       (_, (_, _, psl_ix)) | psl_ix < psl -> IndexToSwap (key, psl) ix
-      _ ->
-        walkDownArr (arr, (ix + 1) `mod` size) (key, psl + PSL 1) size
+      _ -> walkDownArr (arr, (ix + 1) `mod` size) (key, psl + PSL 1) size
+
+-- # Instances
+--------------------------------------------------
+
+instance Consumable (HashMap k v) where
+  consume :: HashMap k v #-> ()
+  consume (HashMap !_ !arr) = lseq arr ()
```

Comment:
You don't need the `!` patterns on the subcomponents:
- The first one is not linear
- The array is linear, but you will consume it anyway, nothing unsafe, so you can follow the types.

```suggestion
  consume (HashMap _ arr) = consume arr
```